### PR TITLE
name in config

### PR DESCRIPTION
**Pull request by BenTheElder**
This was supposed to be a small PR to enable #1412.

On the upside it's not much new code, and we delete a bunch of junk.

What actually happened in order to make this sane:
- drop v1alpha3 config
- eliminate the internal cluster / provider "context" thing (this had it's own name defaulting...)
- move a few things around
  - ListInternalNodes to nodeutils
  - Some of create cluster error handling on nodes already existing / logging => inside create.Cluster impl
- refactor to just pass a config object in more places instead of (name, config)

TODO:
- [ ] support config in `kind delete cluster`
- [ ] document
- [x] add to config testdata

/hold

### CODE REVIEW

File: pkg/cluster/provider.go [hunk 1]
```diff
@@ -36,6 +36,14 @@ import (
 // DefaultName is the default cluster name
 const DefaultName = constants.DefaultClusterName
 
+// defaultName is a helper that given a neame defaults it if unset
```
Comment (aojea) [inline pkg/cluster/provider.go:39]: typo s/neame/name/

Comment (BenTheElder) [reply pkg/cluster/provider.go]: fixed

**Comment by BenTheElder** [comment]
I think we can actually punt documenting for later when this is more settled, but this refactor cleans / unlocks the ability to clean up a LOT so i'm going to rebase it and go ahead

**Comment by BenTheElder** [comment]
It's that kubectl test flake.
/retest

**Pull request merged**